src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -1040,6 +1076,14 @@ void replaceParent(SuiteResult parent) {
         this.parent = parent;
     }
 
+    public @CheckForNull List<FlakyFailure> getFlakyFailures() {
```

Comment:
The `@CheckForNull` is not required now. 

I wonder, why the field itself still is nullable? All those ugly `if (x != null)` guards could be eliminated by working with an empty collection. You can add a readResolve method to ensure that the field is set in old serializations.